### DISCUSSION TITLE
Can't build due to missing file

### DISCUSSION THREAD

**Discussion by Tyri0n** [robertodoering/harpy#822] [General]
Sorry for the noob question, but how exactly do I get this file?

import 'package:go_router/src/go_route_match.dart'

It's nowhere to be found in any subfolders in %LocalAppData%\Pub\Cache\hosted\pub.dev\go_router-4.5.1, or in other versions of go_router that I've checked.

Thanks

**Comment by robertodoering** [comment]
Looks like you have a newer go_router version than specified in the `pubspec.lock`. Did you run `flutter pub upgrade` perhaps? 
Either way, make sure you have no local changes in `pubspec.lock` and run `flutter pub get`.

**Comment by Tyri0n** [comment]
I cloned into a new folder and tried again.

Made the same change in pubspec.yaml:  _material_color_utilities: '>=0.1.5 <= 0.2.0'_

to get around this pub get fail:

> Because every version of flutter_test from sdk depends on material_color_utilities 0.2.0 and harpy depends on material_color_utilities ^0.1.5, flutter_test from sdk is forbidden.
> So, because harpy depends on flutter_test from sdk, version solving failed.
> pub get failed
> command: "C:\dev\Dart\flutter\bin\cache\dart-sdk\bin\dart __deprecated_pub --color --directory . get --example"
> pub env: {
>   "FLUTTER_ROOT": "C:\dev\Dart\flutter",
>   "PUB_ENVIRONMENT": "flutter_cli:get",
>   "PUB_CACHE": "C:\Users\dev\AppData\Local\Pub\Cache",
> }
> exit code: 1
> last line of pub output: "So, because harpy depends on flutter_test from sdk, version solving failed."
> 

then after 'flutter pub run build_runner build' I have 131 issues, these are the errors:

> Target of URI doesn't exist: 'package:go_router/src/go_route_match.dart'.
> Try creating the file referenced by the URI, or try using a URI for a file that does exist.
> 
> The name 'GoRouteMatch' isn't a type, so it can't be used as a type argument.
> Try correcting the name to an existing type, or defining a type named 'GoRouteMatch'.
> 
> The argument type 'List<RouteBase>' can't be assigned to the parameter type 'List<GoRoute>'.
> 
> The name 'GoRouteMatch' isn't a type, so it can't be used as a type argument.
> Try correcting the name to an existing type, or defining a type named 'GoRouteMatch'.
> 

> The name 'ModalBottomSheetRoute' is defined in the libraries 'package:flutter/src/material/bottom_sheet.dart (via package:flutter/material.dart)' and 'package:modal_bottom_sheet/src/bottom_sheet_route.dart (via package:modal_bottom_sheet/modal_bottom_sheet.dart)'.
> Try using 'as prefix' for one of the import directives, or hiding the name from all but one of the imports.
> 
> 
> 

Thanks

**Comment by Tyri0n** [comment]
Looks like 'flutter pub get' is uprading go_router to version 4.1.5 in the lock file.

**Comment by Tyri0n** [comment]
Removing ^ from go_router: ^4.0.0 has fixed that issue.

Just this error remaining:

> The name 'ModalBottomSheetRoute' is defined in the libraries 'package:flutter/src/material/bottom_sheet.dart (via package:flutter/material.dart)' and 'package:modal_bottom_sheet/src/bottom_sheet_route.dart (via package:modal_bottom_sheet/modal_bottom_sheet.dart)'.
> Try using 'as prefix' for one of the import directives, or hiding the name from all but one of the imports

Cheers

**Reply by robertodoering** [reply]
>Try using 'as prefix' for one of the import directives, or hiding the name from all but one of the imports

`import 'package:modal_bottom_sheet/modal_bottom_sheet.dart' as mbs;`
`mbs.ModalBottomSheetRoute`

**Reply by Tyri0n** [reply]
Many thanks
